### PR TITLE
Edit/Delete Submissions and Comments

### PR DESCRIPTION
**Pull request by Brobin**
A lot of this logic was basically the same for submissions and comments, so I decided to put it in the BasePage. 

I cherry picked some code from #66 Thanks to @yskmt

### CODE REVIEW

File: rtv/page.py [hunk 1]
```diff
@@ -251,6 +252,70 @@ def login(self):
         else:
             show_notification(self.stdscr, ['Welcome {}'.format(username)])
 
+    @BaseController.register('d')
+    def delete(self):
+        """
+        Delete a submission or comment.
+        """
+        data = self.content.get(self.nav.absolute_index)
+        if data['author'] != self.reddit.user.name:
```
Comment (michael-lazar) [inline rtv/page.py]: data['author'] may not be a valid field

File: rtv/page.py [hunk 2]
```diff
@@ -251,6 +252,70 @@ def login(self):
...
+            show_notification(self.stdscr, ['You can\'t edit this'])
+            return
+
+        if data['type'] == 'Submission':
+            subreddit = self.reddit.get_subreddit(self.content.name)
+            sub = str(subreddit).split('/')[2]
+            sub_file = '{content}' + SUBMISSION_FILE
```
Comment (michael-lazar) [inline rtv/page.py]: I think it would be better to have this be the default ``SUBMISSION_FILE`` template, and set ``content=''`` in the other case.

Comment (Brobin) [inline rtv/page.py]: sounds good

File: rtv/page.py (hunk 1 above)
Comment (Brobin) [inline rtv/page.py]: I was finding a `KeyError` when I ran it

**Comment by Brobin** [comment]
Overall question, do you think it would be best to just do nothing if a user tries to edit or delete something that is not theirs or not editable? I feel like we have a lot of extra error notifications.

File: rtv/page.py (hunk 2 above)
Comment (Brobin) [inline rtv/page.py]: fixed

**Comment by michael-lazar** [comment]
Merged, thanks @Brobin and @yskmt for the contributions! I fixed a few edge cases, for example you shouldn't be able to delete the submission that you're currently viewing. I also factored out the common error handling between the post/edit/delete functions.

**Pull request merged**